Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
Yeah that's a good point.

On Mon, Apr 4, 2016 at 3:54 PM, Francis Ricci <notifications@github.com>
wrote:

> In Sources/Architecture/ARM/SoftwareBreakpointManager.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r58462758>:
>
> > @@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
> >  }
> >
> >  ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
> > -                                         size_t size) {
> > +                                         size_t size, Mode mode) {
> > +  if (mode != kModeExec)
>
> I think the issue with doing that is that you could theoretically have
> software watchpoints (gdb does), and so I'm not sure that it's really an
> assertion-level failure to try to create one, even if we don't support them
> currently?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/2b7add545bce75bd539f7f39cb938aa59689d8ae#r58462758>
>



-- 
Stephane Sezer